### DISCUSSION TITLE
01_PyTorch Workflow - 42. timestamp 5:35:00 - Problem with plot_prediction

### DISCUSSION THREAD

**Discussion by VELTAX** [mrdbourke/pytorch-deep-learning#341] [Q&A]
Hi, i need some help, here my problem:
when trying to create the prediction graphic, i run into a error:
![image](https://user-images.githubusercontent.com/37710072/224113347-62b1d86b-0903-452a-99cc-67b0efdc14bc.png).

I understand that the problem is that my x and Y are not the same, so i tried to search why, and couldnt find, i have all the same value as in the video, but get more prediction from model_0(X_test)
![image](https://user-images.githubusercontent.com/37710072/224113755-ec7418c6-fa8e-4b9c-8ebe-48e3705e7e50.png).

**Comment by sudokit** [comment]
Can you send the model?

**Reply by VELTAX** [reply]
Here the model:
![image](https://user-images.githubusercontent.com/37710072/224126146-56e66a77-5bc7-4e59-af2d-d0fd9a90f980.png)

**Reply by sudokit** [reply]
Hmm thought u would have a wrong output shape... Weird.

**Comment by TiagoNFC** [comment]
The problem seems to be in :

`def forward(self, x: X_test) -> torch.tensor` 

it should read:

`def forward(self, X: torch.tensor) -> torch.tensor`

**Comment by VELTAX** [comment]
i've found the problem ome timz ago and forgot to close this discussion, the problem was in the model at:
 return self.weights * X + self.bias the X was in upper-cas should have been in lower case : return self.weights * x + self.bias
